pkg/settings/setting.go

```diff
@@ -413,6 +420,18 @@ func NewSetting(name, def string) Setting {
 	return s
 }
 
+// NewSettingWithDefaultOnUpgrade creates a new server setting, taking into account a regular Default value
+// and a Default value to be used on upgrades.
+func NewSettingWithDefaultOnUpgrade(name, def, defOnUpgrade string) Setting {
```

Comment:
Nit: If you want to still use the original `NewSetting` function, you could make a `WithDefaultOnUpgrade(setting Setting, defOnUpgrade string)` which allows you to mutate the setting after it's been made. That way the original function stays the same and we don't have a bunch of "NewSetting" functions if we ever need to enhance this in the future.

Reply:
I completely agree, I had just thought introducing options may not fit the occasion for this PR, but now I think it's worth to do now.